zenoh-kotlin/src/commonMain/kotlin/io/zenoh/Session.kt

```diff
@@ -378,6 +378,39 @@ class Session private constructor(private val config: Config) : AutoCloseable {
         }, handler.receiver(), complete)
     }
 
+    /**
+     * Declare a [Querier].
+     *
+     * A querier allows to send queries to a queryable.
+     *
+     * Queriers are automatically undeclared when dropped.
+     *
+     * Example:
+     * ```kotlin
+     * val session = Zenoh.open(config).getOrThrow();
+     * val keyExpr = "a/b/c".intoKeyExpr().getOrThrow();
+     *
+     * val querier = session.declareQuerier(keyExpr).getOrThrow();
+     * querier.get(callback = {
+     *         it.result.onSuccess { sample ->
+     *             println(">> Received ('${sample.keyExpr}': '${sample.payload}')")
+     *         }.onFailure { error ->
+     *             println(">> Received (ERROR: '${error.message}')")
+     *         }
+     *     }
+     * )
+     * ```
+     */
+    fun declareQuerier(
+        keyExpr: KeyExpr,
+        target: QueryTarget = QueryTarget.BEST_MATCHING,
+        qos: QoS = QoS.default(),
+        consolidation: ConsolidationMode = ConsolidationMode.NONE,
```

Comment:
The default consolidation mode should be AUTO, if it is available in Kotlin.

Reply:
Oh yes, that's a mistake on my part. I'll change that.